t/rose-task-run/44-bunch-incremental.t

```diff
@@ -0,0 +1,139 @@
+# * Run task with --flow=new from outside the N=1 window. Check that this
+#   run of the task runs all subtasks (1 fail, 1 OK).
+#-------------------------------------------------------------------------------
+. $(dirname $0)/test_header
+
+tests 12
+
+get_reg
+
+APP_LOG_PATH="${HOME}/cylc-run/${FLOW}/log/job/2069/bunchapp"
+SCHD_LOG_PATH="${HOME}/cylc-run/${FLOW}/log/scheduler/log"
+
+
+ensure_trigger() {
+    # Trigger and check workflow log for trigger actually working:
+    local jobno="${1}"
+    local stub="2069/bunchapp submitted job:0"
+    searchterm="${stub}${jobno}"
+    while ! grep "${searchterm}" "${SCHD_LOG_PATH}"; do
+        if [[ "jobno" -eq "4" ]]; then
+            cylc trigger "${FLOW}//2069/bunchapp/" --flow=new
+        else
+            cylc trigger "${FLOW}//2069/bunchapp/"
+        fi
+        sleep 2
+    done
+}
+
+
+grep_incrementals() {
+    # Check that a rose app-run runs, skips, and fails the desired number of
+    # Tasks;
+    local TEST_SET="$1"
+    local OK="$2"
+    local FAIL="$3"
+    local SKIP="$4"
+    file_grep "${TEST_SET}-ok" "OK: ${OK}" "${TEST_SET}/job.out"
+    file_grep "${TEST_SET}-fail" "FAIL: ${FAIL}" "${TEST_SET}/job.out"
+    file_grep "${TEST_SET}-skip" "SKIP: ${SKIP}" "${TEST_SET}/job.out"
+}
+
+mkdir -p app/bunchapp
+touch "rose-suite.conf"
+
+# Create a Cylc Workflow
+cat > "flow.cylc" <<__HERE__
+[scheduler]
+    cycle point format = %Y
+
+[scheduling]
+    initial cycle point = 2069
+
+    [[graph]]
+        P1Y = bunchapp
+
+[runtime]
+    [[bunchapp]]
+        script = """
+            # Run a rose app if this is the first, second or fourth submit of 2069, else pass
+            if [[ "\${CYLC_TASK_CYCLE_POINT}" == "2069" ]]; then
+                if [[ "\${CYLC_TASK_SUBMIT_NUMBER}" < 3 || "\${CYLC_TASK_SUBMIT_NUMBER}" > 3 ]]; then
+                    rose task-run
+                else
+                    true
+                fi
+            else
+                true
+            fi
+        """
+
+__HERE__
+
+# App with one bunch member which will fail and one which will pass:
+cat > "app/bunchapp/rose-app.conf" <<__HERE__
+mode=rose_bunch
+meta=rose_bunch
+
+[bunch-args]
+arg1=true false
+
+[bunch]
+command-format=%(arg1)s
+incremental=true
+fail-mode=continue
+__HERE__
+
+# install and play the workflow
+TEST_SET="play-once"
+run_pass "${TEST_SET}" \
+    cylc vip . --host=localhost --workflow-name "${FLOW}" --no-run-name
+
+# Wait for first run of the task to finish - bunch should have one
+# success and one failure:
+poll ! grep TOTAL "${APP_LOG_PATH}/01/job.out" 2>&1 /dev/null
+grep_incrementals "${APP_LOG_PATH}/01" 1 1 0
+
+# Re trigger task in same flow - bunch should skip previously succeeded tasks:
+ensure_trigger "2"
+poll ! grep TOTAL "${APP_LOG_PATH}/02/job.out" 2>&1 /dev/null
+grep_incrementals "${APP_LOG_PATH}/02" 0 1 1
+
+# Trigger the task a third time which doesn't trigger rose bunch
+# Allowing the N=1 window to move on.
```

Comment:
This seems to be demonstrably correct - should it be?

Reply:
By "which doesn't trigger rose bunch" do you mean that rose_bunch doesn't re-run and commands because they have all succeeded?

Reply:
No, in this version of the test the task used CYLC_FLOW_SUBMIT_NUMBER to decide whether to rose app-run or to `true`.